### ISSUE TITLE
text/html representation of DataFrames includes style tag

### ISSUE THREAD

**Issue by rsmith31415** [ipython/ipython#8607]
I noticed that when executing the value of a DataFrame, there is a nice text/html representation, but unfortunately, it includes a style tag with some arbitrary information:

```
In [11]: kc.get_iopub_msg(timeout=1)
Out[11]:
{'buffers': [],
 'content': {u'data': {u'text/html': u'<div style="max-height:1000px;max-width:1500px;overflow:auto;">\n<table border="1" class="dataframe">\n  <thead>\n    <tr style="text-align: right;">\n      <th></th>\n      <th>x</th>\n      <th>y</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>3</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>4</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>5</td>\n      <td>5</td>\n    </tr>\n    <tr>\n      <th>6</th>\n      <td>6</td>\n      <td>6</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>7</td>\n      <td>7</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>8</td>\n      <td>8</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>9</td>\n      <td>9</td>\n    </tr>\n  </tbody>\n</table>\n</div>',
   u'text/plain': u'   x  y\n0  0  0\n1  1  1\n2  2  2\n3  3  3\n4  4  4\n5  5  5\n6  6  6\n7  7  7\n8  8  8\n9  9  9'},
  u'execution_count': 1,
  u'metadata': {}},
 'header': {u'date': datetime.datetime(2015, 7, 4, 15, 9, 36, 92803),
  u'msg_id': u'945c1481-9929-44db-91a8-1e58e0249178',
  u'msg_type': u'execute_result',
  u'session': u'4ad939f9-a88b-42e7-b790-9bc763ae5a66',
  u'username': u'user',
  u'version': u'5.0'},
 'metadata': {},
 'msg_id': u'945c1481-9929-44db-91a8-1e58e0249178',
 'msg_type': u'execute_result',
 'parent_header': {u'date': datetime.datetime(2015, 7, 4, 15, 9, 34, 298826),
  u'msg_id': u'7523ed93-c7d5-4f5d-9ea5-6466ab2e7cd2',
  u'msg_type': u'execute_request',
  u'session': u'f8b5d514-f86c-45af-8060-7d48fef09229',
  u'username': u'user',
  u'version': u'5.0'}}
```

Would it be possible to remove that tag? Is there a workaround to keep the HTML output without the style tag?

**Comment by Carreau** [comment]
That's seem like a Panda question, it is panda that create the html, not IPython.
You could register your own display formatter for the dataframe type, but I'm not sure it will take precedence. Maybe Monkey-patch Pandas ?

**Comment by takluyver** [comment]
I think that may already have been changed in the latest version of pandas.

**Comment by rsmith31415** [comment]
Oh, sorry. I thought it was added by the Jupyter/IPython kernel. I will ask them about it. Thanks.

**Comment by rsmith31415** [comment]
@takluyver Hopefully. I will take a look at the latest version.

**Comment by Carreau** [comment]
> Oh, sorry. I thought it was added by the Jupyter/IPython kernel. I will ask them about it. Thanks.

There are a few "exceptions" but mostly each library define the representation of their object. 
the DataFrame object most likely define a `_repr_html_(self)` methods that generate the HTML.

**Comment by rsmith31415** [comment]
@Carreau Excellent. 

@takluyver  You're right. This is "fixed" in the latest version 16.2:

```
In [13]: kc.get_iopub_msg(timeout=1)
Out[13]:
{'buffers': [],
 'content': {u'data': {u'text/html': u'<div>\n<table border="1" class="dataframe">\n  <thead>\n    <tr style="text-align: right;">\n      <th></th>\n      <th>x</th>\n      <th>y</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>3</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>4</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>5</td>\n      <td>5</td>\n    </tr>\n    <tr>\n      <th>6</th>\n      <td>6</td>\n      <td>6</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>7</td>\n      <td>7</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>8</td>\n      <td>8</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>9</td>\n      <td>9</td>\n    </tr>\n  </tbody>\n</table>\n</div>',
   u'text/plain': u'   x  y\n0  0  0\n1  1  1\n2  2  2\n3  3  3\n4  4  4\n5  5  5\n6  6  6\n7  7  7\n8  8  8\n9  9  9'},
  u'execution_count': 1,
  u'metadata': {}},
 'header': {u'date': datetime.datetime(2015, 7, 4, 15, 35, 13, 139174),
  u'msg_id': u'a44c988b-c529-4f30-9cce-c1a04a11cc80',
  u'msg_type': u'execute_result',
  u'session': u'04605d4b-3492-4418-a57e-7fdd9e474130',
  u'username': u'user',
  u'version': u'5.0'},
 'metadata': {},
 'msg_id': u'a44c988b-c529-4f30-9cce-c1a04a11cc80',
 'msg_type': u'execute_result',
 'parent_header': {u'date': datetime.datetime(2015, 7, 4, 15, 35, 12, 657749),
  u'msg_id': u'8a0e34b2-4bdd-4905-840b-3ae4478dd891',
  u'msg_type': u'execute_request',
  u'session': u'd6c11a5a-0344-4946-b97d-690a222e61a4',
  u'username': u'user',
  u'version': u'5.0'}}
```
